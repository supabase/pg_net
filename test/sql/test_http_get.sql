-- Async
begin;

    with request as (
        select
            net.http_get('http://supabase.io', async:=true) as id
    )
    select
        net.http_collect_response(id)
    from
        request;

rollback;

-- Sync
-- begin;
    -- with request as (
        -- select net.http_get('http://supabase.io', async:=false) as id
    -- )
    -- select
        -- net.http_collect_response(id)
    -- from
        -- request;
-- rollback;

-- TODO run these tests:
-- request with invalid protocol
-- select net.http_get('net://supabase.io');
-- request with invalid hostname
-- select net.http_get('http://new.ycombinator.com');
-- request with NULL body and content_type
-- select net.http_get('http://wikipedia.org');
