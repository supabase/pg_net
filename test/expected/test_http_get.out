-- Async
begin;
    with request as (
        select
            net.http_get('http://supabase.io', async:=true) as id
    )
    select
        net.http_collect_response(id)
    from
        request;
 http_collect_response 
-----------------------
 
(1 row)

rollback;
-- Sync
-- begin;
    -- with request as (
        -- select net.http_get('http://supabase.io', async:=false) as id
    -- )
    -- select
        -- net.http_collect_response(id)
    -- from
        -- request;
-- rollback;
-- TODO run these tests:
-- request with invalid protocol
-- select net.http_get('net://supabase.io');
-- request with invalid hostname
-- select net.http_get('http://new.ycombinator.com');
